git_machete/github.py

```diff
@@ -104,282 +163,296 @@ def get_token_from_gh() -> Optional[GithubTokenAndTokenProvider]:
         stderr = proc.stderr.decode()
         match = re.search(r"Token: (\w+)", stderr)
         if match:
-            return GithubTokenAndTokenProvider(token=match.group(1),
-                                               token_provider=f'auth token for {domain} from `hub` GitHub CLI')
-        return None
+            self.__value = match.group(1)
+            self.__provider = f'auth token for {self.__domain} from `hub` GitHub CLI'
 
-    def get_token_from_hub() -> Optional[GithubTokenAndTokenProvider]:
-        home_path: str = str(Path.home())
-        config_hub_path: str = os.path.join(home_path, ".config", "hub")
-        if os.path.isfile(config_hub_path):
-            with open(config_hub_path) as config_hub:
-                # ~/.config/hub is a yaml file, with a structure similar to:
-                #
-                # {domain1}:
-                # - user: {username1}
-                #   oauth_token: *******************
-                #   protocol: {protocol}
-                #
-                # {domain2}:
-                # - user: {username2}
-                #   oauth_token: *******************
-                #   protocol: {protocol}
-                found_host = False
-                for line in config_hub.readlines():
-                    if line.rstrip() == domain + ":":
-                        found_host = True
-                    elif found_host and line.lstrip().startswith("oauth_token:"):
-                        result = re.sub(' *oauth_token: *', '', line).rstrip().replace('"', '')
-                        return GithubTokenAndTokenProvider(token=result,
```

Comment:
Whoops another problem introduced with this PR... there's a change in logic: now the loop continues after the oauth token is found, which might leads to picking token for another host